### PR TITLE
chore(site): anchor jump don't work

### LINKED ISSUE #45241: [Doc] 官网点击哈希链接不会跳转
### Reproduction link

[https://ant-design.antgroup.com/components/form-cn](https://ant-design.antgroup.com/components/form-cn)

### Steps to reproduce

V5 官方文档，点击组件文档内的链接，不会跳到对应的内容。

image

### What is expected?

点击组件文档内的哈希链接，应该跳到对应内容

### What is actually happening?

点击组件文档内的哈希链接，没有跳到对应内容

| Environment | Info |
| --- | --- |
| antd | 5.9.4 |
| Browser | Chrome 116.0.5845.110 |

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #45241

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    --
| 🇨🇳 Chinese |       修复 dumi 的 ts 类型错误     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8692f70</samp>

Improved TypeScript support and code quality for the `CodePreviewer` component. This component renders live examples of components in the documentation theme.

### <samp>🤖 Generated by Copilot at 8692f70</samp>

* Fix import path of `AntdPreviewerProps` to match renamed file `Previewer.tsx` ([link](https://github.com/ant-design/ant-design/pull/45243/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL11-R11))
* Assert `description` prop to be non-null and cast `jsx` prop to `string` to avoid TypeScript errors in `CodePreviewer.tsx` ([link](https://github.com/ant-design/ant-design/pull/45243/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL174-R174), [link](https://github.com/ant-design/ant-design/pull/45243/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL208-R208))
* Type `dependencies` variable as `Record<PropertyKey, string>` to improve TypeScript inference in `CodePreviewer.tsx` ([link](https://github.com/ant-design/ant-design/pull/45243/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL208-R208))

### CODE REVIEW

**Review by kiner-tang — APPROVED** [review]

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx [hunk 1]
```diff
@@ -171,7 +171,7 @@ const CodePreviewer: React.FC<AntdPreviewerProps> = (props) => {
   });
 
   const localizedTitle = title;
-  const introChildren = <div dangerouslySetInnerHTML={{ __html: description }} />;
+  const introChildren = <div dangerouslySetInnerHTML={{ __html: description! }} />;
```
Comment (afc163) [inline .dumi/theme/builtins/Previewer/CodePreviewer.tsx:174]: 能不加 ! 就不加。

**Comment by afc163** [comment]
标题和内容不符？

File: .dumi/theme/layouts/DocLayout/index.tsx [hunk 2]
```diff
@@ -52,7 +53,14 @@ const DocLayout: React.FC = () => {
...
   }, []);
 
+  // handle hash change or visit page hash from Link component, and jump after async chunk loaded
   useEffect(() => {
+    const id = hash.replace('#', '');
+
+    if (id) document.getElementById(decodeURIComponent(id))?.scrollIntoView();
+  }, [loading, hash]);
+
+  React.useEffect(() => {
```
Comment (afc163) [inline .dumi/theme/layouts/DocLayout/index.tsx:63]:
```suggestion
  useEffect(() => {
```

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx (hunk 1 above)
Comment (afc163) [reply .dumi/theme/builtins/Previewer/CodePreviewer.tsx:174]: description ? <div dangerouslySetInnerHTML={{ __html: description }} /> : null

File: .dumi/theme/layouts/DocLayout/index.tsx [hunk 3]
```diff
@@ -52,6 +53,13 @@ const DocLayout: React.FC = () => {
     }
   }, []);
 
+  // handle hash change or visit page hash from Link component, and jump after async chunk loaded
+  useEffect(() => {
+    const id = hash.replace('#', '');
+
+    if (id) document.getElementById(decodeURIComponent(id))?.scrollIntoView();
```
Comment (li-jia-nan) [inline .dumi/theme/layouts/DocLayout/index.tsx:60]:
```suggestion
    if (id) {
      document.getElementById(decodeURIComponent(id))?.scrollIntoView();
    }
```

Comment (Wxh16144) [reply .dumi/theme/layouts/DocLayout/index.tsx:60]: 以前就这样的，我 revert 就不动逻辑了，后续好追踪变更逻辑~

**Pull request merged**